lib/SPIRV/libSPIRV/SPIRVBasicBlock.cpp

```diff
@@ -67,6 +67,11 @@ SPIRVBasicBlock::addInstruction(SPIRVInstruction *I,
   I->setParent(this);
   if (InsertBefore) {
     auto Pos = std::find(InstVec.begin(), InstVec.end(), InsertBefore);
+    // If insertion of a new instruction before the one passed to the function
```

Comment:
I think we should an an assertion that it is okay to dereference `Pos`.
It returns `InstVec.end()` if search failed

Reply:
Done, this is now checked in the "if" statement

Reply:
Since we are moving the iterator one step back anyway, it is not necessary to check for `Pos != InstVec.end()`
I think ti would be more readable if we replace `(Pos - 1)` by `std::prev(Pos)`

Reply:
If Pos contains InstVec.end(), this means that the InsertBefore instruction was not found in the Vector, and further checks are obsolete anyway. But OK if this improves readability.